src/ucs/arch/cpu.h

```diff
@@ -98,6 +101,11 @@ typedef struct ucs_cpu_builtin_memcpy {
 #  include "ppc64/cpu.h"
 #elif defined(__aarch64__)
 #  include "aarch64/cpu.h"
+#elif defined(__riscv)
+#  ifdef HAVE___CLEAR_CACHE
```

Comment:
What is the problem with built-in clear cache ?

Reply:
It doesn't necessarily clear the cache in the manner expected.  The particular implementation of RISC-V that we're testing on predates the ratification of the CMOBase set of extensions (Zicbom, Zicbop, Zicboz) that implement explicit cache flush operations.  Given that the community was well aware of the forthcoming CMOBase infrastructure (ratified in Nov 2021), the toolchain and system libs have somewhat lagged their support of particular builtins.  They're "implemented", but not fully implemented.  This is quite literally a situation where we're running on rather early hardware/software.